imdclient/IMDClient.py

```diff
@@ -129,9 +131,11 @@ def stop(self):
         if self._multithreaded:
             if not self._stopped:
                 self._buf.notify_consumer_finished()
+                self._wait(0)
                 self._disconnect()
                 self._stopped = True
         else:
+            self._wait(0)
```

Comment:
If sending wait fails, disconnect won't be called. I think we need some kind of try-except-finally statement here. But I'm glad you thought of this edge case, if the client fails out the requested wait behavior should still apply I think